voidcrew/code/modules/overmap/ships/pvp/subverter.dm

```diff
@@ -0,0 +1,329 @@
+	. = FALSE
+	var/can_sub_flag = can_subvert(target_ship)
+	bark_processing(can_sub_flag, target_ship)
+	if (!considered_valid_target(can_sub_flag))
+		return
+	COOLDOWN_START(src, subverter_cooldown, sub_recharge)
+	its_gettin_hot_in_here()
+	//use_power(active_power_usage)
+	if (!target_ship.run_antivirus())
+		force_dock(target_ship)
+		addtimer(CALLBACK(target_ship, /obj/structure/overmap/ship/simulated/.proc/systems_restored), COOLDOWN_TIMELEFT(target_ship, engine_cooldown))
+		return TRUE
+	else
+		target_ship.most_recent_helm.say("Viral agent blocked. Source: [ship.name]")
+
+/**
+*	Update the machine's stats depending on the parts
+*/
+/obj/machinery/subverter/proc/update_stats()
+	sub_recharge = SUBVERTER_RECHARGE_TIME
+	sub_engine = SUBVERTER_ENGINE_STALL_TIME
+	active_power_usage = 1000
+	for(var/obj/item/stock_parts/micro_laser/laser in component_parts)
+		sub_engine += (laser.rating-1) * (15 SECONDS)
+	for(var/obj/item/stock_parts/manipulator/manip in component_parts)
+		sub_recharge -= (manip.rating-1) * (5 SECONDS)
+	for(var/obj/item/stock_parts/capacitor/capac in component_parts)
+		active_power_usage /= capac.rating
+		spiciness /= capac.rating
+
+/obj/machinery/subverter/RefreshParts()
+	update_stats()
+
+/**
+*	Causes the air around the subverter to heat up
+*/
+/obj/machinery/subverter/proc/its_gettin_hot_in_here()
+	var/turf/sub_location = get_turf(src)
+	var/datum/gas_mixture/env = sub_location.return_air()
+	env.set_temperature(env.return_temperature()+spiciness)
+	air_update_turf()
+
+/**
+*	Manual subverting for those who can't afford an auxiliary console
+*
+*	state - Whether or not the subverter got hacked
+*/
+/obj/machinery/subverter/proc/adjust_hacked(state)
+	hacked = state
+	if (!hacked)
+		return
+	if (ship.prefix == "NEU")
+		subvert_bark("NEU Vessels cannot use the subverter!")
+		return
+	//find nearest non-allied pvp ship and attempt to subvert
+	var/obj/structure/overmap/ship/simulated/nearest
+	var/ship_min_dist
+	var/can_sub_flag
+	for (var/obj/structure/overmap/ship/simulated/ship_in_view in view(ship.sensor_range, get_turf(ship)))
+		can_sub_flag = can_subvert(ship_in_view)
+		if (ship_in_view == ship || !considered_valid_target(can_sub_flag))
+			continue
+		if (!nearest)
+			nearest = ship_in_view
+			ship_min_dist = get_dist(ship, nearest)
+		else
+			var/ship_dist = get_dist(ship, ship_in_view)
+			if (ship_dist < ship_min_dist)
+				nearest = ship_in_view
+				ship_min_dist = ship_dist
+	if (nearest)
+		attempt_to_subvert(nearest)
+	else
+		subvert_bark("No subvertable vessels detected.")
+
+//Normal machine hacking stuff
+/obj/machinery/subverter/proc/shock(mob/user, prb)
+	if(machine_stat & (BROKEN|NOPOWER))
+		return FALSE
+	if(!prob(prb))
+		return FALSE
+	var/datum/effect_system/spark_spread/spark = new /datum/effect_system/spark_spread
+	spark.set_up(5, 1, src)
+	spark.start()
+	if (electrocute_mob(user, get_area(src), src, 0.7, TRUE))
+		return TRUE
+	else
```

Comment:
this else is redundant

Reply:
lol i just copied that from the autholathe.dm